components/brave_wallet_ui/components/desktop/views/nfts/components/nfts.tsx

```diff
@@ -129,6 +131,17 @@ export const Nfts = (props: Props) => {
     return filteredNfts.sort((a, b) => a.name.localeCompare(b.name))
   }, [filteredNfts])
 
+  // effects
+  React.useEffect(() => {
+    const getAutoDisoveryStatus = () => {
+      getNftDiscoveryEnabledStatus()
+        .then(enabled => setIsNftAutoDiscoveryEnabled(enabled))
+        .catch(error => console.error(error))
+    }
+
+    getAutoDisoveryStatus()
+  }, [getNftDiscoveryEnabledStatus])
```

Comment:
once `getNftDiscoveryEnabledStatus` is moved to the api slice, you can get the value using the generated hook
```suggestion
  // queries
  const { data: isNftAutoDiscoveryEnabled } = useGetNftDiscoveryEnabledStatusQuery()
```